azurelinuxagent/common/dhcp.py

```diff
@@ -80,23 +79,30 @@ def wireserver_route_exists(self):
         This is true when running in a virtual network.
         :return: True if a route to KNOWN_WIRESERVER_IP exists.
         """
-        route_exists=False
+        route_exists = False
         logger.info("test for route to {0}".format(KNOWN_WIRESERVER_IP))
         try:
-            route = shellutil.run_get_output("grep -c {0} /proc/net/route".format(KNOWN_WIRESERVER_IP_ENTRY))
+            # grep -c returns 1 if the count is 0, so ignore the return code
+            route = shellutil.run_get_output(
```

Comment:
I feel strongly that scripting languages (like python) that handle grep-like functionality in a three-line function should not be shelling out to grep. There are multiple reasons for this:
- performance
- reduced memory footprint
- reduced attack surface
- semantic transparency ("What does grep -c do?" vs count=CountMatchingLinesInFile(filename))